Include migration from Salt 3000 to Bundle in troubleshooting guide (#1903)
uyuni-project/uyuni-docs: modules/client-configuration/pages/tshoot-clients.adoc

====
Increasing this value will cause {productname} Server to take longer to check if a minion is unreachable or unresponsive, causing the {productname} Server to be slower or less responsive overall.
====

== Salt 3000 to Salt Bundle Migration

=== Switch {sles} 12 or {centos} 7 minions (Salt 3000 EOL) to Salt Bundle.

.Procedure: Switching with [literal]``util.mgr_switch_to_venv_minion`` state to [package]``venv-salt-minion``

. Apply [literal]``util.mgr_switch_to_venv_minion`` with no pillar specified first.

>   This will result in the switch to [package]``venv-salt-minion`` with copying configuration files etc.

Reviewer: ```suggestion
  This will result in the switch to [package]``venv-salt-minion`` with copying configuration files and customization.
```
I hope we can use this wording to avoid the obscure "etc.".

Author: I believe this was meant to be "in etc" but I'm not absolutely sure. This was copied from https://documentation.suse.com/suma/4.3/en/suse-manager/client-configuration/contact-methods-saltbundle.html#_switching_from_salt_minion_to_salt_bundle
@agraul it this correct?

Reviewer 2: I don't know. That wording was introduced in https://github.com/uyuni-project/uyuni-docs/pull/1458/files as far as I can tell. No one commented on it, so I can't speak to the authors intention.

Reviewer: Thanks for digging it out (I was involved, but do not remember any details).  Maybe, @meaksh know whether the "/etc" directory was meant or other things ("etc.") -- if the latter, for example what?  Maybe, we can simply delete the etc here?  Pablo, I hope you can help!

Reviewer: looks fixed now.

  It will not clean up the original [package]``salt-minion`` configurations and its packages.
+
----
salt <minion_id> state.apply util.mgr_switch_to_venv_minion
----
. Apply [literal]``util.mgr_switch_to_venv_minion`` with [literal]``mgr_purge_non_venv_salt`` set to [literal]``True`` to remove [package]``salt-minion`` and with [literal]``mgr_purge_non_venv_salt_files`` set to [literal]``True`` to remove all the files related to [package]``salt-minion``.

> This second step ensures the first step was processed, and then removes the old configuration files and the now obsolete [package]``salt-minion`` package.

Reviewer: ```suggestion
  This second step ensures the first step was processed, and then removes the old configuration files and the now obsolete [package]``salt-minion`` package.
```
FYI: The indentation helps translation related tools with segmentation.

+
----
salt <minion_id> state.apply util.mgr_switch_to_venv_minion pillar='{"mgr_purge_non_venv_salt_files": True, "mgr_purge_non_venv_salt": True}'
----

=== Switch {sles} 12 or {centos} 7 SSH minions (Salt 3000 EOL) to Salt Bundle.

.Procedure: Enabling the Salt Bundle with Salt SSH support

> . Make sure that the bootstrap repository is created

Reviewer: ```suggestion
. Make sure that the bootstrap repository is created.
```

> The bootstrap repository must be created before bootstrapping the client with {webui}. Salt SSH is using the Salt Bundle taken from the bootstrap repository based on the detected target operating system.

Reviewer 2: The transition for Salt SSH from `py27-compat-salt` / Salt 3000 to the Salt Bundle is not only about bootstrapping, it's for all Salt SSH uses that target SLE12/RHEL7. I wonder how this part will be understood, I imagine readers will skip over the paragraph after reading the first sentence.

Maybe we can leave the bootstrapping with webui part out and just say something like "create the bootstrap repo with the venv-salt-minion package for SLE12/RHEL7, it's required for Salt SSH with the bundle" (with better wording)

Author: Maybe it's clearer now!

> The bootstrap repository must be created before bootstrapping the client with {webui}. Salt SSH is using the Salt Bundle taken from the bootstrap repository based on the detected target operating system.
> For more information, see xref:client-configuration:bootstrap-repository.adoc#_prepare_to_create_a_bootstrap_repository[Prepare to Create a Bootstrap Repository]

Reviewer: ```suggestion
  The bootstrap repository must be created before bootstrapping the client with {webui}.
  Salt SSH is using the Salt Bundle taken from the bootstrap repository based on the detected target operating system.
  For more information, see xref:client-configuration:bootstrap-repository.adoc#_prepare_to_create_a_bootstrap_repository[Prepare to Create a Bootstrap Repository].
```

> . Include Salt Bundle in the bootstrap repository

Reviewer: ```suggestion
. Include Salt Bundle in the bootstrap repository.
```

> The Salt Bundle must also be included to the bootstrap repository relevant to the client by calling [command]``mgr-create-bootstrap-repo`` explicitly specifying [package]``venv-salt-minion`` as an additional package. 

Reviewer 2: The way I read this, it sounds like one has to create the bootstrap repo twice. AFAIK just using the below command is enough, can we consolidate this and the previous sections?

+
----
mgr-create-bootstrap-repo -c DISTRIBUTION_LABEL venv-salt-minion

> ----

Reviewer: ```suggestion
----

```

> . Specify in [path]``/etc/rhn/rhn.conf``

Reviewer: ```suggestion
. Specify in [path]``/etc/rhn/rhn.conf``:
```

+
----
web.ssh_salt_pre_flight_script = /usr/share/susemanager/salt-ssh/preflight.sh

web.ssh_use_salt_thin = false
----
. Create the extra drop-in configuration file for Salt Master [path]``/etc/salt/master.d/ssh-preflight.conf`` with [literal]``ssh_run_pre_flight`` set to [literal]``true``:
+
----
ssh_run_pre_flight: true

> ----

Reviewer: ```suggestion
----

```
